test/iron_behavior_active_state_test.dart

```diff
@@ -0,0 +1,194 @@
+              expect(activeTarget.getAttribute('active'), isNotNull);
+              expect(activeTarget.getAttribute('aria-checked'), isNotNull);
+              expect(activeTarget.getAttribute('aria-checked'), 'true');
+              done.complete();
+            } catch (e) {
+              done.complete(e);
+            }
+          });
+          return done.future;
+        });
+
+        test('the aria attribute is updated correctly', () {
+          var done = new Completer();
+          activeTarget.ariaActiveAttribute = 'aria-checked';
+          downAndUp(activeTarget, () {
+            try {
+              expect(activeTarget.getAttribute('active'), isNotNull);
+              expect(activeTarget.getAttribute('aria-checked'), isNotNull);
+              expect(activeTarget.getAttribute('aria-checked'), 'true');
+
+              activeTarget.ariaActiveAttribute = 'aria-pressed';
+              expect(activeTarget.getAttribute('aria-checked'), isNull);
+              expect(activeTarget.getAttribute('aria-pressed'), isNotNull);
+              expect(activeTarget.getAttribute('aria-pressed'), 'true');
+              done.complete();
+            } catch (e) {
+              done.complete(e);
+            }
+          });
+          return done.future;
+        });
+      });
+    });
+
+    group('without toggles attribute', () {
+      group('when mouse is down', () {
+        test('does not get an active attribute', () {
+          expect(activeTarget.getAttribute('active'), isNull);
+          down(activeTarget);
+          expect(activeTarget.getAttribute('active'), isNull);
+        });
+      });
+
+      group('when mouse is up', () {
+        test('does not get an active attribute', () {
+          down(activeTarget);
+          expect(activeTarget.getAttribute('active'), isNull);
+          up(activeTarget);
+          expect(activeTarget.getAttribute('active'), isNull);
+        });
+      });
+    });
+
+    group('when space is pressed', () {
+      test('triggers a click event', () {
+        var done = new Completer();
+        activeTarget.on['click'].take(1).listen((_) {
+          done.complete();
+        });
+        pressSpace(activeTarget);
+        return done.future;
+      });
+
+      test('only triggers click after the key is released', () {
+        var done = new Completer();
+        var keyupTriggered = false;
+
+        activeTarget.on['keyup'].take(1).listen((_) {
+          keyupTriggered = true;
+        });
+
+        activeTarget.on['click'].take(1).listen((_) {
+          try {
+            expect(keyupTriggered, true);
+            done.complete();
+          } catch (e) {
+            done.complete(e);
+          }
+        });
+
+        pressSpace(activeTarget);
+        return done.future;
+      });
+    });
+
+    group('when enter is pressed', () {
+      test('triggers a click event', () {
+        var done = new Completer();
+        activeTarget.on['click'].take(1).listen((_) {
```

Comment:
BTW, you can probably get rid of a lot of the completers here. For example, many of these `take(1).listen`, could be converted into `take(1).single` or `.first`:

```dart
var future = activeTarget.onClick.take(1).single;
pressEnter(activeTarget);
return future;
```